hypha/apply/funds/views.py

```diff
@@ -1579,6 +1583,45 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
+class SubmissionWithdrawView(SingleObjectTemplateResponseMixin, BaseDetailView):
+    model = ApplicationSubmission
+    success_url = reverse_lazy("funds:submissions:list")
+    template_name_suffix = "_confirm_withdraw"
+
+    def post(self, request, *args, **kwargs):
+        return self.withdraw(request, *args, **kwargs)
+
+    def withdraw(self, request, *args, **kwargs):
+        if not settings.ENABLE_SUBMISSION_WITHDRAWAL:
+            raise PermissionDenied
+
+        if not request.user.is_applicant:
+            raise PermissionDenied
```

Comment:
Can we use `UserPassesTestMixin` for this check? I think that makes the code easier to read.

Reply:
Or if `perform_transition` function does this check we do not need it all all as you mention below.